tasks/main.yml

```diff
@@ -107,7 +107,8 @@
   vars:
     reboot_message: rebooting for robertdebock/ansible-role-update
   when:
-    - update_update_all_software_apk.changed or
+    - update_reboot and
```

Comment:
From `defaults/main.yml`:

```
# When updating systems, a reboot may be required. Here you can select to:
# "yes": Always reboot when packages have changed.
# "no": Never reboot when packages have changed.
```

So if I read the current conditions, it does what it should do: reboot when things have changed AND when `update_reboot` is set to `yes`.

(But please correct me if I'm wrong...)

Reply:
Sorry, my bad, as far as I can tell the role in it's current state does exactly what you intended it to do and using the original condition `    - update_reboot | bool` at the end avoids the warning in Ansible 2.8. I was looking for a nice way to prevent all reboots but that would require removing the dependency, replacing it with a conditional include and so on and now I'm thinking it might be better to work around the role the way it is, cause, well, it's really beautiful. Thanks for taking the time to review this.